src/main/java/seedu/address/model/statistics/Statistics.java

```diff
@@ -0,0 +1,342 @@
+     * @return A table containing the differences in the two tables of basic statistics
+     */
+    private ArrayList<ArrayList<Object>> secondMinusFirst(ArrayList<ArrayList<Object>> firstTable,
+                                                          ArrayList<ArrayList<Object>> secondTable) {
+        ArrayList<ArrayList<Object>> result = createEmptyTableWithoutPercentage();
+        int rowSize = firstTable.size();
+        int columnSize = firstTable.get(0).size();
+        for (int i = 0; i < rowSize; i++) {
+            ArrayList<Object> tableEntry = result.get(i);
+            for (int j = 0; j < columnSize; j++) {
+                Object firstInput = firstTable.get(i).get(j);
+                Object secondInput = secondTable.get(i).get(j);
+                tableEntry.set(j, handleInputType(firstInput, secondInput));
+            }
+        }
+        return result;
+    }
+
+    /**
+     * Handles the interaction between the first input and the second input by their type
+     * @param firstInput The input from the first table
+     * @param secondInput The input from the second table
+     * @return The correct result to be placed into the result table
+     */
+    private Object handleInputType(Object firstInput, Object secondInput) {
+        if (firstInput instanceof Tag) {
+            return firstInput;
+        } else if (firstInput instanceof Double) {
+            return (Double) secondInput - (Double) firstInput;
+        } else if (firstInput instanceof Integer) {
+            return (Integer) secondInput - (Integer) firstInput;
+        } else {
+            return null;
+        }
+    }
+
+    /**
+     * Calculates the basic statistics between 2 dates and stores the results in the statsBuilder field
+     * @param startDate The starting date
+     * @param endDate The ending date
+     */
+    private void basicStats(Timestamp startDate, Timestamp endDate) {
+        requireNonNull(startDate);
+        requireNonNull(endDate);
+
+        ArrayList<ArrayList<Expense>> data = extractRelevantExpenses(startDate, endDate);
+
+
+        statsBuilder.append("Statistics Summary\n");
+        statsBuilder.append(String.format("From %s to %s\n", startDate, endDate));
+
+
+        ArrayList<ArrayList<Object>> table = createEmptyTableWithPercentage();
+
+        List<String> headers = List.of("Category: ",
+                "Amount Spent ($): ",
+                "Number of entries: ",
+                "Percentage: ");
+        convertDataToFigures(data, table, true);
+        convertTableToString(table, headers);
+    }
+
+    /**
+     * Creates an empty table
+     *
+     * @return An list of table entries
+     */
+    private ArrayList<ArrayList<Object>> createEmptyTableWithPercentage() {
+        ArrayList<ArrayList<Object>> table = new ArrayList<>();
+        for (int i = 0; i < tags.size() + 1; i++) {
+            ArrayList<Object> tableEntry = new ArrayList<>();
```

Comment:
Consider creating a class `StatisticData` for potential data types, rather than using `Object`.